Library/Homebrew/uninstall.rb

```diff
@@ -96,18 +97,27 @@ def uninstall_kegs(kegs_by_rack, force: false, ignore_dependencies: false, named
       end
     end
 
-    def handle_unsatisfied_dependents(kegs_by_rack, ignore_dependencies: false, named_args: [])
+    def handle_unsatisfied_dependents(kegs_by_rack, casks: [], ignore_dependencies: false, named_args: [])
       return if ignore_dependencies
 
       all_kegs = kegs_by_rack.values.flatten(1)
-      check_for_dependents(all_kegs, named_args: named_args)
+      check_for_dependents(all_kegs, casks: casks, named_args: named_args)
     rescue MethodDeprecatedError
       # Silently ignore deprecations when uninstalling.
       nil
     end
 
-    def check_for_dependents(kegs, named_args: [])
-      return false unless result = Keg.find_some_installed_dependents(kegs)
+    def check_for_dependents(kegs, casks: [], named_args: [])
+      result_kegs = Keg.find_some_installed_dependents(kegs)
+      result_casks = Keg.find_cask_dependents(kegs, casks)
+
+      return false if result_kegs.blank? && result_casks.blank?
+
+      result = if result_kegs.present? && result_casks.present?
+        [(result_kegs[0] + result_casks[0]).uniq, result_kegs[1] + result_casks[1]]
```

Comment:
The `[0]` and `[1]` here is a bit hard to follow here. Maybe use intermediate variables so it's more obvious what's going on here?